Using videojs-wavesurfer with angular 8 without webpack

## Description
I want to use videojs-wavesurfer in my angular project without a webpack.

## Steps to reproduce
I have implemented the help of the [Example](https://collab-project.github.io/videojs-wavesurfer/#/angular)  and I am getting error as I have not used webpack.

## Error output
ERROR in ./node_modules/videojs-wavesurfer/dist/videojs.wavesurfer.js
Module not found: Error: Can't resolve 'WaveSurfer' in 'C:\Lex_UI\fusion\node_modules\videojs-wavesurfer\dist'

ERROR in ./node_modules/videojs-wavesurfer/dist/videojs.wavesurfer.js
Module not found: Error: Can't resolve 'videojs' in 'C:\Lex_UI\fusion\node_modules\videojs-wavesurfer\dist'
i ?wdm?: Failed to compile.

## Results

### Actual
If we modify 
**videojs** to **video.js** and
**WaveSurfer** to **WaveSurfer.js**  
in the videojs.wavesurfer.js file which is in the node module, it is working.

But this, not the best approach.

Please suggest me the alternative approach for doing this, or can do these changes in the actual code so that if we install this library we don't have to modify it manually.

## Versions
videojs-wavesurfer:- 3.2.0

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,13 @@
 # videojs-wavesurfer changelog
 
+
+## 4.0.0 - unreleased
+
+- Fixed webpack externals configuration: it's no longer needed to use additional webpack
+  configuration in React/Angular projects (#109)
+- Bump required wavesurfer.js version to 4.0.1 or newer
+
+
 ## 3.2.0 - 2020/05/19
 
 **Backwards-incompatible changes** (when upgrading from a previous version):

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,7 +1,7 @@
 # videojs-wavesurfer changelog
 
 
-## 4.0.0 - unreleased
+## 3.3.0 - unreleased
 
 - Fixed webpack externals configuration: it's no longer needed to use additional webpack
   configuration in React/Angular projects (#109)

--- a/build-config/fragments/common.js
+++ b/build-config/fragments/common.js
@@ -49,11 +49,22 @@ module.exports = {
     },
     // specify dependencies for the library that are not resolved by webpack,
     // but become dependencies of the output: they are imported from the
-    // environment during runtime.
-    externals: [
-        {'video.js': 'videojs'},
-        {'wavesurfer.js': 'WaveSurfer'}
-    ],
+    // environment during runtime and never directly included in the
+    // videojs-record library.
+    externals: {
+        'video.js': {
+            commonjs: 'video.js',
+            commonjs2: 'video.js',
+            amd: 'video.js',
+            root: 'videojs' // indicates global variable
+        },
+        'wavesurfer.js': {
+            commonjs: 'wavesurfer.js',
+            commonjs2: 'wavesurfer.js',
+            amd: 'wavesurfer.js',
+            root: 'WaveSurfer' // indicates global variable
+        }
+    },
     module: {
         rules: [
             {

--- a/docs/angular.md
+++ b/docs/angular.md
@@ -17,7 +17,12 @@ dependencies:
     "name": "angular-videojs-wavesurfer",
     "version": "1.0.0",
     "scripts": {
-        "start": "webpack-dev-server --mode development"
+        "ng": "ng",
+        "start": "ng serve",
+        "build": "ng build",
+        "test": "ng test",
+        "lint": "ng lint",
+        "e2e": "ng e2e"
     },
     "dependencies": {
         "@angular/common": "^9.1.6",
@@ -67,42 +72,6 @@ Create `tsconfig.json`:
         "experimentalDecorators": true,
         "target": "ES5"
     }
-}
-```
-
-Create a Webpack configuration file called `webpack.config.js`:
-
-```javascript
-const HtmlWebpackPlugin = require('html-webpack-plugin');
-const ProvidePlugin = require('webpack/lib/ProvidePlugin');
-
-module.exports = {
-    entry: './src/main.ts',
-    resolve: {
-        extensions: ['.ts', '.js'],
-        alias: {
-            videojs: 'video.js',
-            WaveSurfer: 'wavesurfer.js'
-        }
-    },
-    module: {
-        rules: [
-            {
-                test: /\.ts$/,
-                use: ['ts-loader']
-            },
-            {
-                test: /\.(html|css)$/,
-                use: 'raw-loader'
-            }
-        ]
-    },
-    plugins: [
-        new ProvidePlugin({
-            videojs: 'video.js/dist/video.cjs.js'
-        }),
-        new HtmlWebpackPlugin({ template: './src/index.html' })
-    ]
 }
 ```
 
@@ -258,15 +227,12 @@ export class AppModule { }
 Create an Angular polyfills file in `src/polyfills.ts`:
 
 ```ts
-import 'core-js/features/reflect';
 import 'zone.js/dist/zone';
 ```
 
 Create the Angular main file in `src/main.ts`:
 
 ```ts
-import './polyfills';
-
 import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 import { AppModule } from './app/app.module';
 
@@ -305,4 +271,4 @@ Start the development server:
 npm start
 ```
 
-And open http://localhost:8080/ in a browser.
+And open http://localhost:4200 in a browser.

--- a/docs/angular.md
+++ b/docs/angular.md
@@ -7,78 +7,49 @@ for Angular.
 
 ## Installation
 
-Create a project directory, e.g. `angular-videojs-wavesurfer`.
-
-Create a `package.json` file inside that project directory that lists the project
-dependencies:
-
-```json
-{
-    "name": "angular-videojs-wavesurfer",
-    "version": "1.0.0",
-    "scripts": {
-        "ng": "ng",
-        "start": "ng serve",
-        "build": "ng build",
-        "test": "ng test",
-        "lint": "ng lint",
-        "e2e": "ng e2e"
-    },
-    "dependencies": {
-        "@angular/common": "^9.1.6",
-        "@angular/compiler": "^9.1.6",
-        "@angular/core": "^9.1.6",
-        "@angular/forms": "^9.1.6",
-        "@angular/platform-browser": "^9.1.6",
-        "@angular/platform-browser-dynamic": "^9.1.6",
-        "@angular/router": "^9.1.6",
-        "core-js": "^3.6.5",
-        "rxjs": "^6.5.5",
-        "zone.js": "^0.10.3"
-    },
-    "devDependencies": {
-        "@types/node": "^13.13.5",
-        "html-webpack-plugin": "^4.3.0",
-        "raw-loader": "^4.0.1",
-        "ts-loader": "^7.0.4",
-        "typescript": "^3.8.3",
-        "webpack": "^4.43.0",
-        "webpack-cli": "^3.3.11",
-        "webpack-dev-server": "^3.11.0"
-    }
-}
-```
-
-Install the dependencies:
+Install the [Angular CLI](https://cli.angular.io) globally:
 
 ```console
-npm install
+npm install -g @angular/cli
+```
+
+Create a new application, e.g. `videojs-wavesurfer-angular`:
+
+```console
+ng new videojs-wavesurfer-angular
 ```
 
 Install `videojs-wavesurfer` and `@types/video.js`:
 
 ```console
+cd videojs-wavesurfer-angular
 npm install --save videojs-wavesurfer @types/video.js
-```
-
-## Configuration
-
-Create `tsconfig.json`:
-
-```json
-{
-    "compilerOptions": {
-        "emitDecoratorMetadata": true,
-        "experimentalDecorators": true,
-        "target": "ES5"
-    }
-}
 ```
 
 ## Application
 
-Create the `src/app/` directories and add a new Angular component for videojs-wavesurfer
-in `src/app/videojs.wavesurfer.component.ts`:
+Create a new Angular component:
+
+```console
+ng generate component videojs-wavesurfer
+```
+
+Replace the content of `src/app/videojs-wavesurfer/videojs-wavesurfer.component.css` with:
+
+```css
+/* change player background color */
+#audio_clip1 {
+    background-color: #42f489;
+}
+```
+
+Replace the content of `src/app/videojs-wavesurfer/videojs-wavesurfer.component.html` with:
+
+```html
+<audio id="audio_{{idx}}" class="video-js vjs-default-skin"></audio>
+```
+
+Replace the content of `src/app/videojs-wavesurfer/videojs-wavesurfer.component.ts` with:
 
 ```ts
 import {
@@ -102,19 +73,11 @@ WaveSurfer.microphone = MicrophonePlugin;
 import * as Wavesurfer from 'videojs-wavesurfer/dist/videojs.wavesurfer.js';
 
 @Component({
-  selector: 'videojs-wavesurfer',
-  template: `
-    <style>
-    /* change player background color */
-    #audio_clip1 {
-         background-color: #42f489;
-    }
-    </style>
-    <audio id="audio_{{idx}}" class="video-js vjs-default-skin"></audio>
-    `
+  selector: 'app-videojs-wavesurfer',
+  templateUrl: './videojs-wavesurfer.component.html',
+  styleUrls: ['./videojs-wavesurfer.component.css']
 })
-
-export class VideoJSWavesurferComponent implements OnInit, OnDestroy {
+export class VideojsWavesurferComponent implements OnInit, OnDestroy {
 
   // reference to the element itself: used to access events and methods
   private _elementRef: ElementRef
@@ -176,7 +139,7 @@ export class VideoJSWavesurferComponent implements OnInit, OnDestroy {
       videojs.log(msg);
 
       // load file
-      this.player.src({src: '/hal.wav', type: 'audio/wav'});
+      this.player.src({src: 'assets/hal.wav', type: 'audio/wav'});
     });
 
     this.player.on('waveReady', event => {
@@ -208,67 +171,30 @@ export class VideoJSWavesurferComponent implements OnInit, OnDestroy {
 }
 ```
 
-Create the Angular app module in `src/app/app.module.ts`:
-
-```ts
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
-
-import { VideoJSWavesurferComponent } from './videojs.wavesurfer.component';
-
-@NgModule({
-    imports: [BrowserModule],
-    declarations: [VideoJSWavesurferComponent],
-    bootstrap: [VideoJSWavesurferComponent]
-})
-export class AppModule { }
-```
-
-Create an Angular polyfills file in `src/polyfills.ts`:
-
-```ts
-import 'zone.js/dist/zone';
-```
-
-Create the Angular main file in `src/main.ts`:
-
-```ts
-import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
-import { AppModule } from './app/app.module';
-
-platformBrowserDynamic().bootstrapModule(AppModule);
-```
-
-And finally, create the main index HTML file in `src/index.html`:
+Replace content of `src/app/app.component.html` with:
 
 ```html
-<!DOCTYPE html>
-<html>
-<head>
-    <base href="/" />
-    <title>Angular videojs-wavesurfer example</title>
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <!-- styles -->
-    <link href="node_modules/video.js/dist/video-js.css" rel="stylesheet">
-    <link href="node_modules/videojs-wavesurfer/dist/css/videojs.wavesurfer.css" rel="stylesheet">
-</head>
-<body>
-    <videojs-wavesurfer></videojs-wavesurfer>
-</body>
-</html>
+<app-videojs-wavesurfer></app-videojs-wavesurfer>
+```
+
+Add the following to `src/styles.css`:
+
+```css
+@import '~video.js/dist/video-js.css';
+@import '~videojs-wavesurfer/dist/css/videojs.wavesurfer.css';
 ```
 
 ## Media
 
 Download the [example audio file](https://github.com/collab-project/videojs-wavesurfer/raw/master/examples/media/hal.wav)
-and place it in the project directory.
+and place it in the `src/assets` directory.
 
 ## Run
 
 Start the development server:
 
 ```console
-npm start
+ng serve
 ```
 
 And open http://localhost:4200 in a browser.

--- a/docs/react.md
+++ b/docs/react.md
@@ -8,64 +8,22 @@ for React.
 
 ## Installation
 
-Create an example React application called `audio-app`:
+Create an example React application called `videojs-wavesurfer-react`:
 
 ```console
-npx create-react-app audio-app
+npx create-react-app videojs-wavesurfer-react
 ```
 
 Install videojs-wavesurfer:
 
 ```console
-cd audio-app
+cd videojs-wavesurfer-react
 npm install --save videojs-wavesurfer
-```
-
-Install [react-app-wired](https://github.com/timarney/react-app-rewired) that we'll
-use to configure Webpack:
-
-```console
-npm install react-app-rewired --save-dev
-```
-
-## Configuration
-
-Create a `config-overrides.js` file in the root directory:
-
-```javascript
-const webpack = require("webpack");
-
-module.exports = function override(config, env) {
-  // Extend the config to work with videojs-wavesurfer without ejecting create react app.
-  // Reference: https://collab-project.github.io/videojs-wavesurfer/#/react
-  const videojsPlugin = new webpack.ProvidePlugin({
-    videojs: "video.js/dist/video.cjs.js"
-  });
-  const videojsAlias = {
-    videojs: "video.js",
-    WaveSurfer: "wavesurfer.js"
-  };
-  config.resolve.alias = { ...config.resolve.alias, ...videojsAlias };
-  config.plugins.push(videojsPlugin);
-  return config;
-};
-```
-
-Change the existing calls to `react-scripts` in the `scripts` section of `package.json`
-for `start`, `build` and `test`:
-
-```json
-"scripts": {
-    "start": "react-app-rewired start",
-    "build": "react-app-rewired build",
-    "test": "react-app-rewired test",
-    "eject": "react-scripts eject"
-}
 ```
 
 ## Application
 
-Edit `src/index.js`:
+Replace content of `src/index.js` with:
 
 ```javascript
 import React from 'react';
@@ -106,7 +64,7 @@ ReactDOM.render(
 serviceWorker.unregister();
 ```
 
-Edit `src/App.js`:
+Replace content of `src/App.js`:
 
 ```javascript
 /* eslint-disable */

--- a/docs/vue.md
+++ b/docs/vue.md
@@ -26,30 +26,6 @@ cd videojs-wavesurfer-app
 npm install --save videojs-wavesurfer
 ```
 
-## Configuration
-
-Create `vue.config.js` with the following content:
-
-```javascript
-const webpack = require('webpack');
-
-module.exports = {
-  configureWebpack: {
-    resolve: {
-        alias: {
-            videojs: 'video.js',
-            WaveSurfer: 'wavesurfer.js'
-        }
-    },
-    plugins: [
-        new webpack.ProvidePlugin({
-            videojs: 'video.js/dist/video.cjs.js'
-        })
-    ]
-  }
-}
-```
-
 ## Application
 
 Create `src/components/VideoJSWavesurfer.vue`:
@@ -86,7 +62,7 @@ Create `src/components/VideoJSWavesurfer.vue`:
                             backend: 'MediaElement',
                             displayMilliseconds: true,
                             debug: true,
-                            waveColor: '#6f7982',
+                            waveColor: '#3b4045',
                             progressColor: 'black',
                             cursorColor: 'black',
                             hideScrollbar: true

--- a/docs/webpack.md
+++ b/docs/webpack.md
@@ -13,7 +13,7 @@ npm install webpack webpack-dev-server webpack-cli css-loader style-loader -D
 Install videojs-wavesurfer:
 
 ```console
-npm install videojs-wavesurfer
+npm install --save videojs-wavesurfer
 ```
 
 ## Configuration
@@ -22,7 +22,6 @@ Create the Webpack config file called `webpack.config.js`:
 
 ```javascript
 const path = require('path');
-const webpack = require('webpack');
 const basePath = path.resolve(__dirname);
 
 module.exports = {
@@ -39,17 +38,6 @@ module.exports = {
         contentBase: basePath,
         watchContentBase: true
     },
-    resolve: {
-        alias: {
-            videojs: 'video.js',
-            WaveSurfer: 'wavesurfer.js'
-        }
-    },
-    plugins: [
-        new webpack.ProvidePlugin({
-            videojs: 'video.js/dist/video.cjs.js'
-        })
-    ],
     module: {
         rules: [{
             test: /\.css$/,

--- a/karma.conf.js
+++ b/karma.conf.js
@@ -172,7 +172,7 @@ module.exports = function(config) {
     };
 
     if (ci) {
-        configuration.browsers = ['Chrome_headless', 'Firefox_headless'];
+        configuration.browsers = ['Firefox_headless'];
         configuration.detectBrowsers.enabled = false;
         configuration.singleRun = true;
 

--- a/package.json
+++ b/package.json
@@ -86,7 +86,7 @@
     "karma-safaritechpreview-launcher": "^2.0.2",
     "karma-verbose-reporter": "0.0.6",
     "karma-webpack": "^4.0.2",
-    "mini-css-extract-plugin": "^0.9.0",
+    "mini-css-extract-plugin": "^0.10.0",
     "node-sass": "^4.14.1",
     "npm-run-all": "^4.1.5",
     "optimize-css-assets-webpack-plugin": "^5.0.3",
@@ -95,11 +95,11 @@
     "rimraf": "^3.0.2",
     "sass-loader": "^9.0.3",
     "style-loader": "^1.2.1",
-    "terser-webpack-plugin": "^4.0.0",
+    "terser-webpack-plugin": "^4.1.0",
     "webpack": "^4.44.1",
     "webpack-cli": "^3.3.12",
     "webpack-dev-server": "^3.11.0",
     "webpack-merge": "^5.1.1",
-    "webrtc-adapter": "^7.6.4"
+    "webrtc-adapter": "^7.7.0"
   }
 }

